doc/did-method-spec.md

```diff
@@ -399,6 +438,84 @@ The DID resolution result for a deactivated DID has the following shape:
 }
 ```
 
+## Metadata
+
+The `resolve` method returns an object with the following properties: `didDocument`, `didDocumentMetadata`,
+`didResolutionMetadata`.
+
+### DID Document Metadata
+
+When resolving a DID document that has had updates, the latest update MUST be listed in the `didDocumentMetadata`.
+* `versionId` MUST be the block height of the latest update.
+* `updated` MUST be the ISO date string of the block time of the latest update.
+
+Example:
+```json
+{
+  "didDocumentMetadata": {
+    "versionId": "12090175",
+    "updated": "2021-03-22T18:14:29Z"
+  }
+}
+```
+
+### DID Resolution Metadata
+
+```json
+{
+  "didResolutionMetadata": {
+    "contentType": "application/did+ld+json"
+  }
+}
+```
+
+## Resolving DID URIs with query parameters.
+
+### `versionId` query string parameter
+
+This DID method supports resolving previous versions of the DID document by specifying a `versionId` parameter.
+
+Example: `did:ethr:0x26bf14321004e770e7a8b080b7a526d8eed8b388?versionId=12090175`
+
+The `versionId` is the block height at which the DID resolution MUST be performed.
+Only ERC1056 events prior to or contained in this block height are to be considered when building the event history.
+
+If there are any events after that block that mutate the DID, the earliest of them SHOULD be used to populate the
+properties of the `didDocumentMetadata`:
+* `nextVersionId` MUST be the block height of the next update to the DID document.
+* `nextUpdate` MUST be the ISO date string of the block time of the next update.
+
+In case the DID has had updates prior to or included in the `versionId` block height, the `updated` and `versionId`
+properties of the `didDocumentMetadata` MUST correspond to the latest block prior to the `versionId` querystring param.
+
+Any timestamp comparisons of `validTo` fields of the event history MUST be done against the `versionId` block timestamp.
+
+Example:
+`?versionId=12101682`
+```json
+{
+  "didDocumentMetadata": {
+    "versionId": "12090175",
+    "updated": "2021-03-22T18:14:29Z",
+    "nextVersionId": "12276565",
+    "nextUpdate": "2021-04-20T10:48:42Z"
+  }
+}
+```
+
+#### Security considerations of DID versioning
+
+Applications must take precautions when using versioned DID URIs.
+If a key is compromised and revoked then it can still be used to issue signatures on behalf of the "older" DID URI.
+The use of versioned DID URIs is only recommended in some limited situations where the timestamp of signatures can also
+be verified, where malicious signatures can be easily revoked, and where applications can afford to check for these
+explicit revocations of either keys or signatures.
+Wherever versioned DIDs are in use, it SHOULD be made obvious to users that they are dealing with potentially revoked data.
+
+### `initial-state` query string parameter
+
+TBD
```

Comment:
should we already include initial-state if not specified?

Reply:
You're right.. Maybe not